### PR TITLE
chore: update azure sdk to support BYOK

### PR DESCRIPTION
**Pull request by andyzhangx**
**Reason for Change**:

chore: update azure sdk to support BYOK

upgrade azure-sdk-for-go version from `v34.1.0` to `v36.2.0`

**Issue Fixed**:

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
```
vi go.mod
go mod vendor
go mod tidy
```

### CODE REVIEW

**Comment by jackfrancis** [comment]
@devigned could you do a sanity check review as well?

**Comment by mboersma** [comment]
lgtm++

File: go.sum [hunk 1]
```diff
@@ -1,6 +1,8 @@
 cloud.google.com/go v0.34.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
 github.com/Azure/azure-sdk-for-go v34.1.0+incompatible h1:uW/dgSzmRQEPXwaRUN8WzBHJy5J2cp8cw1ea908uFj0=
 github.com/Azure/azure-sdk-for-go v34.1.0+incompatible/go.mod h1:9XXNKU+eRnpl9moKnB4QOLf1HestfXbmab5FXxiDBjc=
```
Comment (devigned) [inline go.sum]: Why are we still referencing this version? Seems like this should have been removed by `go mod tidy`.

Comment (andyzhangx) [reply go.sum]: good catch, I have used `go mod tidy` to fix this

**Pull request merged**